core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -42,10 +44,10 @@ object Consumer {
   }
 
   trait CommittableOffset extends Committable {
-    def offset: Offset
+    def offset: PartitionOffset
   }
 
-  final case class Offset(topic: String, partition: Int, offset: Long)
+  final case class PartitionOffset(topic: String, partition: Int, offset: Long)
 
   object CommittableOffsetBatch {
```

Comment:
Why are we keeping implementation in internal package? We can just build map `(topic, partition) -> msg` and update some elements based on user rule. Next on commit we should commit all messages in this map. Looks like there is no dependency on implementation.